scripts/image

```diff
@@ -259,6 +259,14 @@ if [ "$1" = "release" -o "$1" = "mkimage" -o "$1" = "amlpkg" -o "$1" = "noobs" ]
 
   RELEASE_DIR="target/$IMAGE_NAME"
 
+  if [ -n "$DEVICE" -a -d "$PROJECT_DIR/$PROJECT/devices/$DEVICE/install" ]; then
+    INSTALL_SRC_DIR="$PROJECT_DIR/$PROJECT/devices/$DEVICE/install"
+  else
+    INSTALL_SRC_DIR="$PROJECT_DIR/$PROJECT/install"
+  fi
+
+  MKIMAGE_TOOL_CMD="$(which mkimage) -A $TARGET_KERNEL_ARCH -O linux -T script -C none -d"
```

Comment:
What's the point of `$(which mkimage)`? If `mkimage` is found on PATH then `mkimage` will work just as well as `$(which mkimage)`.

Note that `$(which mkimage)` will return nothing if `mkimage` is not installed.

If you add `u-boot-tools` (for Ubuntu, other distros may vary) to `scripts/checkdeps` (but only when building u-boot that needs `mkimage`) then the build won't even start unless `mkimage` is available - this would be the correct fix (IMHO), as well as removing `$(which ...)`.

Reply:
Also, you're setting MKIMAGE_TOOL_CMD for all projects, even though `mkimage` is only needed for uboot builds.

I'd suggest moving this within `scripts/mkimage` - pass TARGET_KERNEL_ARCH instead to `scripts/mkimage` as that's the only variable component, and then run the `mkimage` command only when it is required.

Or do as @Kwiboo suggests, which might be cleaner (haven't checked).

Reply:
The idea of calling `$(which mkimage)` is to return a full path to system tool and not call `scripts/mkimage`. Perhaps this is not needed?

I will also add a commit to require u-boot-tools or add a package.

Reply:
Will apply the second comment, thank you!